src/backup.ts

```diff
@@ -0,0 +1,387 @@
+   * const database = spanner.database('my-database');
+   * const backup = instance.backup('my-backup');
+   * const alreadyExists = await backup.exists();
+   * console.log(`Does backup exist? ${alreadyExists}`);
+   */
+  async exists(): Promise<boolean> {
+    try {
+      // Attempt to read metadata to determine whether backup exists
+      await this.getMetadata();
+      // Found therefore it exists
+      return true;
+    } catch (err) {
+      if (err.code === status.NOT_FOUND) {
+        return false;
+      }
+      // Some other error occurred, rethrow
+      throw err;
+    }
+  }
+
+  updateExpireTime(expireTime: PreciseDate): Promise<Backup>;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback: UpdateExpireTimeCallback
+  ): void;
+  /**
+   * Sets the expiry time of a backup.
+   *
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#updateExpireTime
+   * @returns {Promise<Backup>} when resolved, the backup's expire time will
+   *     have been updated.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const myBackup = instance.backup('my-backup');
+   * const oneDay = 1000 * 60 * 60 * 24;
+   * const newExpireTime = new PreciseDate(Date.now() + oneDay);
+   * await myBackup.updateExpireTime(newExpireTime);
+   */
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback?: UpdateExpireTimeCallback
+  ): void | Promise<Backup> {
+    this.expireTime = expireTime;
+
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IUpdateBackupRequest = {
+      backup: {
+        name: this.formattedName_,
+        expireTime: expireTime.toStruct(),
+      },
+      updateMask: {
+        paths: ['expire_time'],
+      },
+    };
+    return this.request<databaseAdmin.spanner.admin.database.v1.IBackup>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'updateBackup',
+        reqOpts,
+      },
+      err => {
+        callback!(err, err ? undefined : this);
+      }
+    );
+  }
+
+  delete(): Promise<void>;
+  delete(callback: DeleteCallback): void;
+  /**
+   * Deletes a backup.
+   *
+   * @method Backup#delete
+   * @returns {Promise<void>} when resolved, the backup will have been deleted.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const myBackup = instance.backup('my-backup');
+   * await myBackup.delete();
+   */
+  delete(callback?: DeleteCallback): void | Promise<void> {
+    const reqOpts: databaseAdmin.spanner.admin.database.v1.IDeleteBackupRequest = {
+      name: this.formattedName_,
+    };
+    return this.request<databaseAdmin.spanner.admin.database.v1.IBackup>(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'deleteBackup',
+        reqOpts,
+      },
+      err => {
```

Comment:
Is there a response here we could send back, too?

Reply:
Unfortunately no, DeleteBackup() returns an empty response.